sharedbuffers/mapped_struct.py

```diff
@@ -341,6 +342,75 @@ def unpack_from(cls, buf, offs, idmap = None):
         key, values = mapped_list.unpack_from(buf, offs, idmap)
         return cls(zip(key, values))
 
+class proxied_buffer(object):
+
+    @classmethod
+    def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0, header_packer=struct.Struct('L')):
+        header_packer.pack_into(buf, offs, len(obj))
+        offs += header_packer.size
+
+        end_offs = offs + len(obj)
+        buf[offs:end_offs] = obj
+
+        return end_offs
+
+    @classmethod
+    def unpack_from(cls, buf, offs, idmap = None, header_packer=struct.Struct('L')):
```

Comment:
You have the struct twice. That's ripe territory for bugs.

You want to have it once, as a class attribute probably.